CONTRIBUTING.md

```diff
@@ -54,6 +53,27 @@ When you are ready to submit your pull-request:
 
 For instructions on submitting a pull-request, please see:  [Using Pull Requests ](https://help.github.com/articles/using-pull-requests/) and [Sending Pull Requests](http://help.github.com/send-pull-requests/).
 
+### Getting your changes reviewed
+
+Once you've submitted your pull request, you want
+other members of the development community to review
+whether integrating your change will cause problems
+for any users or the maintainability of the software.
+
+If you have an idea who might be able to spot such issues
+in the parts of the code and functionality affected by your changes,
+notify them by requesting a review using the **Reviewers** menu
+to the right of the summary you just wrote
+and/or `@`-mentioning them in a comment.
+
+Reviewers may request you to rephrase or adjust things
+before they allow the changes to be integrated.
+If they do, commit the amendments as new, separate changes,
+to allow the reviewers to see what changed since they last read your code.
+Do not overwrite previously-reviewed commits with
+ones that include additional changes (by `--amend`ing or squashing)
+until the reviewers approve.
+
```

Comment:
I would add something saying that multi-commit PRs will typically be squashed during merge into a single commit, and that the commit messages will be combined / simplified by the person doing the merge. For longer development efforts with rounds of change requests and adjustments, it would be appreciated if someone doing the development provided a "final form" commit message to be used with the squashed commit.
This way the developer who worked on a feature will know that during squashing there will be editing of their commit messages.

Reply:
I have not encountered this, while my pull requests nearly always have multiple commits representing distinct concerns within the reviewed proposal. But I can imagine that it can sometimes be convenient for reviewers to squash review-based amendment commits themselves instead of asking the requester to do so.

Reply:
Does this already happen, maybe even without consulting the requester? I do agree that it makes sense to add a sentence here telling contributors to expect it.

Reply:
I added a sentence to this pull request about expecting requests/offers to squash amendment commits.